Microsoft.Toolkit.Uwp.Services/Services/Storage/LocalStorageService.cs

```diff
@@ -0,0 +1,103 @@
+using Windows.Storage;
+
+namespace Microsoft.Toolkit.Uwp.Services.Services.Storage
+{
+    /// <summary>
+    /// Store data in the Local environment (only on the current device)
+    /// </summary>
+    public class LocalStorageService
+    {
+        private ApplicationDataContainer localSettings = ApplicationData.Current.LocalSettings;
+        private StorageFolder localFolder = ApplicationData.Current.LocalFolder;
+        private JsonSerializer serializer = new JsonSerializer();
+
+        /// <summary>
+        /// Retrieve single item by its key
+        /// </summary>
+        /// <typeparam name="T">Type of object retrieved</typeparam>
+        /// <param name="key">Key of the object</param>
+        /// <returns>The T object</returns>
+        public T Retrieve<T>(string key)
+        {
+            string value = (string)localSettings.Values[key];
+            if (value != null)
+            {
+                return JsonConvert.DeserializeObject<T>(value);
+            }
+
+            return default(T);
+        }
+
+        /// <summary>
+        /// Save string value by its key
+        /// </summary>
+        /// <param name="key">Key of the value saved</param>
+        /// <param name="value">Value to save</param>
+        public void Save(string key, string value)
+        {
+            localSettings.Values[key] = value;
+        }
+
+        /// <summary>
+        /// Save single item by its key
+        /// </summary>
+        /// <typeparam name="T">Type of object saved</typeparam>
+        /// <param name="key">Key of the value saved</param>
+        /// <param name="value">Object to save</param>
+        public void Save<T>(string key, T value)
+        {
+            Save(key, JsonConvert.SerializeObject(value));
+        }
+
+        /// <summary>
+        /// Retrieve object from file
+        /// </summary>
+        /// <typeparam name="T">Type of object retrieved</typeparam>
+        /// <param name="filePath">Path to the file that contains the object</param>
+        /// <returns>Waiting task until completion with the object in the file</returns>
+        public async Task<T> RetrieveFileAsync<T>(string filePath)
+        {
+            var file = await localFolder.CreateFileAsync(filePath, CreationCollisionOption.OpenIfExists);
+            if (file != null)
+            {
+                string value = await FileIO.ReadTextAsync(file);
+                if (value != null)
+                {
+                    return JsonConvert.DeserializeObject<T>(value);
+                }
+            }
+
+            return default(T);
+        }
+
+        /// <summary>
+        /// Save extended string inside file
+        /// </summary>
+        /// <param name="filePath">Path to the file that will contain the object</param>
+        /// <param name="value">Value to save</param>
+        /// <returns>Waiting task until completion</returns>
+        public async Task SaveFileAsync(string filePath, string value)
```

Comment:
I think the @ScottIsAFool concern is about simplicity. Do users need it? We try to avoid adding features for the sake of adding them.
I do not say that your code is bad or useless. I'm just questioning the fact that we need it in the toolkit